### ISSUE TITLE
About page `solana` and `broadside` templates break when content has a figure with a caption

### ISSUE THREAD

**Issue by CeresBarros** [quarto-dev/quarto-cli#9356]
### Bug description

I have an About page with several listings and a sidebar with a logo that appears unformatted when I render the website locally using the current pre-release version and when the website is rendered/built using `quarto-dev/quarto-actions/setup@v2` with defaults.

If I render locally using the CRAN version of Quarto (I see `packageVersion("quarto")` being "1.4"), the formatting seems okay

### Steps to reproduce

`about.qmd` is the "problematic page" in the repo below.
https://github.com/PredictiveEcology/PredictiveEcology.org

(note that rendered website files are on `gh-pages` branch)

### Expected behavior

Page should look like this (as it does with CRAN version) 
![image](https://github.com/quarto-dev/quarto-cli/assets/22214224/23b451eb-cb61-448b-8195-b6b0ef632ade)

### Actual behavior

Page looks like this
![image](https://github.com/quarto-dev/quarto-cli/assets/22214224/4e04b537-82a9-46f4-9142-8b5f2fbe32fb)

### Your environment

RStudio 2023.12.1+402 

Windows 10

### Quarto check output

problematic versions are the pre-release below and the version used by `quarto-dev/quarto-actions/setup@v2`
```
Quarto 1.5.29
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.29
      Path: C:\Users\cbarros\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Tex:  (not detected)

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

[>] Checking R installation...........OK
      Version: 4.3.3
      Path: C:/Users/cbarros/AppData/Local/Programs/R/R-4.3.3
      LibPaths:
        - C:/Users/cbarros/AppData/Local/Programs/R/R-4.3.3/library
      knitr: 1.46
      rmarkdown: 2.26

[>] Checking Knitr engine render......OK
```

working version:

```
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.3.450
      Path: C:\Program Files\RStudio\resources\app\bin\quarto\bin
      CodePage: 1252

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

[>] Checking R installation...........OK
      Version: 4.3.3
      Path: C:/Users/cbarros/AppData/Local/Programs/R/R-4.3.3
      LibPaths:
        - C:/Users/cbarros/AppData/Local/Programs/R/R-4.3.3/library
      knitr: 1.46
      rmarkdown: 2.26

[>] Checking Knitr engine render......OK

```

**Comment by CeresBarros** [comment]
please note that I'm currently working around this issue by using 
```
      - name: Set up Quarto
        uses: quarto-dev/quarto-actions/setup@v2
        with:
          version: 1.3.450
```
in the `.github/workflows/publish.yml` GHA workflow

**Comment by mcanouil** [comment]
The R package has nothing to do with the issue or the version of Quarto CLI.

The `quarto` R package, is only calling Quarto CLI, which run outside of R, not in R.

This being said, would you mind trying to share a small reproducible example rather than a complete website?

**Comment by CeresBarros** [comment]
Thanks for trying our repo @mcanouil. 

I haven't had time to make a reprex, I'm afraid. I will try ASAP.

To be clear, in my original post/tests I was able to get the formatting right when using Quarto from RStudio after having installed _only_ from CRAN. When I installed the **pre-release** version from the Quarto website the formatting was messed up, as it was when GHActions rendered and published the site (which I believe is using v1.4.515?).

I will try again with the current Quarto version you pointed to. 

Nevertheless, there still seems to be something going on between versions v1.3.450 and v1.4.515, since forcing `quarto-dev/quarto-actions/setup@v2` to use v1.3.450 fixes the formatting issue I'm seeing.

**Comment by mcanouil** [comment]
> after having installed only from CRAN

The R package is unrelated, you can install or update it, it does not have any effect on Quarto CLI.
Quarto is a CLI, not an R package.
The R package is only there to call Quarto CLI from R.
The R package is not needed at all, no matter if you installed it from CRAN or GitHub.

RStudio comes with a version of Quarto CLI.

**Comment by CeresBarros** [comment]
I understand now -- I thought the Quarto R packages was installing Quarto CLI in the background. Thanks for clarifying that.

Quick update: I just installed v.4.553 and re-rendered. The formatting issue is still present:

```
C:\Users\cbarros\GitHub\PredictiveEcology.org> quarto check
Quarto 1.4.553
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.4.553
      Path: C:\Users\cbarros\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Tex:  (not detected)

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

[>] Checking R installation...........OK
      Version: 4.3.3
      Path: C:/Users/cbarros/AppData/Local/Programs/R/R-4.3.3
      LibPaths:
        - C:/Users/cbarros/AppData/Local/Programs/R/R-4.3.3/library
      knitr: 1.46
      rmarkdown: 2.26

[>] Checking Knitr engine render......OK
```

![image](https://github.com/quarto-dev/quarto-cli/assets/22214224/9c5bf1e8-c498-4bcd-8057-3687ff2e179a)


Will work on a reprex

**Comment by mcanouil** [comment]
Could you try 1.5 pre-release?

**Comment by CeresBarros** [comment]
Here's a reprex:

1. Create a blog template
```
quarto create project blog reprex
```

2. Change `about.qmd` to:
```
 ---
title: "Home"
comments: false
about: 
  id: about-pec
  template: solana
  links: 
    - icon: globe
      text: a link
      href: https://predictiveecology.org/
    - icon: github
      text: GitHub
      href: https://github.com/PredictiveEcology
listing: 
  - id: alist
    contents: posts
    sort: 
      - "author"
    fields: [image, title, description]
    max-description-length: 1000000
    image-height: 250px
    image-align: left
    sort-ui: false
    filter-ui: false
    categories: false
    feed: false
    page-size: 100
---

::: {#about-pec}

Lorem ipsicum 

#### a subtitle

![a pic](https://quarto.org/quarto.png){width="250" style="float:right; padding:0px 0px 10px 50px;"}

some text some text some text

:::

------------------------------------------------------------------------

### A list

::: {#alist}
:::

```

3. change `_quarto.yml` to:
```
project:
  type: website

website:
  title: "Reprex"
  navbar:
    logo: "https://quarto.org/quarto.png"
    right:
      - about.qmd
      - icon: github
        href: https://github.com/
      - icon: twitter
        href: https://twitter.com
  sidebar:
    logo: "https://quarto.org/quarto.png"
format:
  html:
    theme: cosmo
    css: styles.css
```

4. run
```
quarto render
```

And navigate to "Home" tab.


With:
```
quarto check
Quarto 1.4.553
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.4.553
      Path: C:\Users\cbarros\AppData\Local\Programs\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Tex:  (not detected)

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....(None)

      Unable to locate an installed version of Python 3.
      Install Python 3 from https://www.python.org/downloads/

[>] Checking R installation...........OK
      Version: 4.3.3
      Path: C:/Users/cbarros/AppData/Local/Programs/R/R-4.3.3
      LibPaths:
        - C:/Users/cbarros/AppData/Local/Programs/R/R-4.3.3/library
      knitr: 1.46
      rmarkdown: 2.26

[>] Checking Knitr engine render......OK
```

I see 
![image](https://github.com/quarto-dev/quarto-cli/assets/22214224/cc7034fb-b1ed-448c-9f8b-6f9eeea4fa9a)

**Comment by CeresBarros** [comment]
> Could you try 1.5 pre-release?

Do mean this one:
https://quarto.org/docs/download/prerelease.html
?

If so, I have (see original post)

**Comment by mcanouil** [comment]
Bisected this to the following commit: 71a64a4ff65d68e2f7feed3f30b8435709f578ea

**Comment by cscheid** [comment]
There are a number of things going wrong here in our styling. They're all best illustrated by videos showing the webpage behavior as the width expands and we hit different responsive breakpoints.

## Behavior from the reported reprex

https://github.com/quarto-dev/quarto-cli/assets/285675/873ebbf8-bed1-4cf2-b886-2941126e8e44

## Behavior after removing the `#about-pec` div

https://github.com/quarto-dev/quarto-cli/assets/285675/2f08a0bd-93ce-42c0-9818-133164d4ae16

**Comment by cscheid** [comment]
The worst bug happens when there's a _figure_ inside the `about` div. An image by itself doesn't trigger the bug, it has to have a caption.

**Comment by cscheid** [comment]
Here's another bug.

This should have failed validation, but didn't (it's missing the `template` field)

````
---
title: "Home"
comments: false
about: 
  id: about-pec
---

::: {#about-pec}

![a pic](https://quarto.org/quarto.png)

:::
````

**Comment by cscheid** [comment]
This minimal example already triggers the bug:

````
---
title: "Home"
comments: false
about: 
  id: about-pec
  template: solana
---

::: {#about-pec}

![a pic](https://quarto.org/quarto.png)

:::
````

https://github.com/quarto-dev/quarto-cli/assets/285675/3e647666-7bbc-4892-a8c1-49919b71c3f6

**Comment by cscheid** [comment]
The final bug is that in the narrowest breakpoint, the about page somehow centers all text:

<img width="707" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/d2e6d971-c28f-4e8f-83fd-6c0d1284e875">

<img width="766" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/5fc88186-962f-4ce8-ba32-a37d7d5476d8">

**Comment by cscheid** [comment]
In the process of figuring this out, I realized that all of these issues only come up with `template: solana` for the about page.

**Comment by CeresBarros** [comment]
thanks all for your work so far on figuring this out! I'll keep my eye out for the bug fixes and in the mean time will keep using v1.3.450 to render the website

**Comment by cscheid** [comment]
I backported this fix to `v1.4`. It will be out in the next v1.4 patch release.
